src/pack.c

```diff
@@ -583,17 +591,11 @@ static int pack_dependency_chain(git_dependency_chain *chain_out,
 		if (type != GIT_OBJECT_OFS_DELTA && type != GIT_OBJECT_REF_DELTA)
 			break;
 
-		base_offset = get_delta_base(p, &w_curs, &curpos, type, obj_offset);
+		error = get_delta_base(&base_offset, p, &w_curs, &curpos, type, obj_offset);
 		git_mwindow_close(&w_curs);
 
-		if (base_offset == 0) {
-			error = packfile_error("delta offset is zero");
-			goto on_error;
-		}
```

Comment:
You're removing this error case here, but as far as I can see it's not been added to `get_delta_base`.

Reply:
Done. Not sure if it's needed in both places that I added. I would assume `git_oidmap_get` would not store an invalid entry, so maybe I am being paranoid.

Reply:
Better be defensive while programming than running into errors because of taking too many things for granted :+1: